src/code/z_sram_NES.c

```diff
@@ -28,58 +28,77 @@ typedef struct PersistentCycleFlags {
 /**
  * Array of bitwise flags which won't be turned off on a cycle reset (will persist between cycles)
  */
-PersistentCycleFlags sPersistentCycleFlags[SCENE_MAX] = {
+PersistentCycleSceneFlags sPersistentCycleSceneFlags[SCENE_MAX] = {
 #include "tables/scene_table.h"
 };
 
 #undef DEFINE_SCENE
 #undef DEFINE_SCENE_UNSET
 
-// TODO: figure out a way to use the WEEKEVENTREG defines here
+#define WEEK_FLG2(flg) (2 << (2 * BIT_FLAG_TO_SHIFT(flg)))
+#define WEEK_FLG(flg) (3 << (2 * BIT_FLAG_TO_SHIFT(flg)))
+
 // weekEventReg flags which will be not be cleared on a cycle reset
-u16 D_801C66D0[ARRAY_COUNT(gSaveContext.save.saveInfo.weekEventReg)] = {
-    /*  0 */ 0xFFFC,
-    /*  1 */ 0xFFFF,
-    /*  2 */ 0xFFFF,
-    /*  3 */ 0xFFFF,
+u16 sPersistentCycleWeekEventRegs[ARRAY_COUNT(gSaveContext.save.saveInfo.weekEventReg)] = {
```

Comment:
We should probably have a note explaining the values of this array "discard" the index part of the `WEEKEVENTREG_` macros, only keeping the mask part. It is very important that the index of each macro matches the index of this array, otherwise different flags will be set on cycle reset.

Reply:
Made a note, let me know if it's clear enough